exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceExporter.java

```diff
@@ -15,134 +15,108 @@
-      Map.of("User-Agent", "opentelemetry-operations-java/" + TraceVersions.EXPORTER_VERSION);
-  private static final HeaderProvider HEADER_PROVIDER = () -> HEADERS;
+  private final TraceConfiguration customTraceConfiguration;
+  private final AtomicReference<SpanExporter> internalTraceExporter;
 
-  public static SpanExporter createWithDefaultConfiguration() throws IOException {
-    TraceConfiguration configuration = TraceConfiguration.builder().build();
-    return TraceExporter.createWithConfiguration(configuration);
+  private TraceExporter(TraceConfiguration configuration) {
+    this.customTraceConfiguration = configuration;
+    this.internalTraceExporter = new AtomicReference<>(null);
   }
 
-  public static SpanExporter createWithConfiguration(TraceConfiguration configuration)
-      throws IOException {
-    String projectId = configuration.getProjectId();
-    TraceServiceStub stub = configuration.getTraceServiceStub();
-
-    // TODO: Remove stub.
-    if (stub == null) {
-      TraceServiceSettings.Builder builder = TraceServiceSettings.newBuilder();
-
-      // We only use the batchWriteSpans API in this exporter.
-      builder
-          .batchWriteSpansSettings()
-          .setSimpleTimeoutNoRetries(
-              org.threeten.bp.Duration.ofMillis(configuration.getDeadline().toMillis()));
-      // For testing, we need to hack around our gRPC config.
-      if (configuration.getInsecureEndpoint()) {
-        builder.setCredentialsProvider(NoCredentialsProvider.create());
-        builder.setTransportChannelProvider(
-            FixedTransportChannelProvider.create(
-                GrpcTransportChannel.create(
-                    ManagedChannelBuilder.forTarget(configuration.getTraceServiceEndpoint())
-                        .usePlaintext()
-                        .build())));
-      } else {
-        Credentials credentials =
-            configuration.getCredentials() == null
-                ? GoogleCredentials.getApplicationDefault()
-                : configuration.getCredentials();
-        builder.setCredentialsProvider(
-            FixedCredentialsProvider.create(checkNotNull(credentials, "credentials")));
-        builder.setEndpoint(configuration.getTraceServiceEndpoint());
-        builder.setHeaderProvider(HEADER_PROVIDER);
-      }
+  private static SpanExporter generateSkeletonTraceExporter(TraceConfiguration config) {
+    return new TraceExporter(config);
+  }
 
-      return new TraceExporter(
-          projectId,
-          new CloudTraceClientImpl(TraceServiceClient.create(builder.build())),
-          configuration.getAttributeMapping(),
-          configuration.getFixedAttributes());
-    }
-    return TraceExporter.createWithClient(
-        projectId,
-        new CloudTraceClientImpl(TraceServiceClient.create(stub)),
-        configuration.getAttributeMapping(),
-        configuration.getFixedAttributes());
+  private SpanExporter createActualTraceExporter() throws IOException {
```

Comment:
Consider inlining this method.

Reply:
Looking at this with fresh eyes, I agree with this. Thanks.